addons/framework/statsTracking/fn_handleMissionEnd.sqf

```diff
@@ -23,114 +23,123 @@ diag_log "[CMF] STATS - fn_handleMissionEnd - Beginning stat tracking reporting
 // Get still alive player data
 [] remoteExec ["CMF_fnc_handleMissionEndLocal", 0];
 
-// Announce to players stats are updating
-"[CMF] Stats will begin updating in 5 seconds!" remoteExec ["systemChat", 0];
-
-// Sleep to allow CMF_fnc_handleMissionEndLocal to propagate and get all users info into the main array
-uiSleep 5;
-
-_numberOfPlayersInDataArray = count playersDataArray;
-
-// Update database values
-{	
-	// Assign variables from hashmap
-	_playerUID				= _x;           // Str | Player Steam ID
-	_roleStatus				= _y select 0;  // Str | String of type of role selected
-	_missionsAttended 		= _y select 1;  // Int | Will always be 1
-	_timeAlive 				= _y select 2;  // Int | Time in seconds player was not a spectator
-	_tvtKills				= _y select 3;  // Int | PvP Kills
-	_tvtDeaths 				= _y select 4;  // Int | PvP Deaths
-	_coopKills				= _y select 5;  // Int | PvAI Kills
-	_coopDeaths				= _y select 6;  // Int | PvAI Deaths
-	_friendlyFireEvents		= _y select 7;  // Int | Dumbass friendly fired one or more friendlies, this is how many they killed	
-	_civsKilled				= _y select 8;  // Int | This is how many non-potato based AI a player has killed
-	_shotsFired				= _y select 9;  // Int | amount of shots fired by player
-	_playerLeftMidMission	= _y select 10; // Int | 1 == true | 0 == false	
-
-	// Log data
-	diag_log format ["[CMF] STATS - fn_handleMissionEnd - Updating stats for steamid: %1",_playerUID];
-	
-	// Update specific sql entries
-	switch _roleStatus do
-	{
-		case "inf_pl_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_pl_roles = inf_pl_roles + 1 WHERE steamid = %1",_playerUID];
-		};
-		case "inf_sl_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_sl_roles = inf_sl_roles + 1 WHERE steamid = %1",_playerUID];
-		};
-		case "inf_ftl_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_ftl_roles = inf_ftl_roles + 1 WHERE steamid = %1",_playerUID];
-		};
-		case "inf_gi_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_gi_roles = inf_gi_roles + 1 WHERE steamid = %1",_playerUID];
-		};
-		case "inf_specialty_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_specialty_roles = inf_specialty_roles + 1 WHERE steamid = %1",_playerUID];
-		};
-		case "inf_medical_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET inf_medical_roles = inf_medical_roles + 1 WHERE steamid = %1",_playerUID]; 
-		};
-		case "pilot_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET pilot_roles = pilot_roles + 1 WHERE steamid = %1",_playerUID]; 
-		};
-		case "tanker_roles":{
-			"extDB3" callExtension format ["0:SQL:UPDATE a3rolestats SET tanker_roles = tanker_roles + 1 WHERE steamid = %1",_playerUID]; 
-		};
-	}; 
+numberOfPlayersInDataArray = count playersDataArray;
+playersComplete = 0;
+
+[] Spawn {
```

Comment:
> This spawn is most likely unnecessary as well. We're already in a spawned thread.

The spawn is to make this script multi-threaded, so as every player in the hashmap is called up, they're subsequently given a spawned thread to do their data logging tasks.

Reply:
Ok, but we spawn a thread with `[] spawn CMF_fnc_handleMissionEnd.sqf`. Then in that thread, we spawn another thread on line 29. Then we spawn a thread per player on the foreach loop. One thread too many imo.

Reply:
> Ok, but we spawn a thread with `[] spawn CMF_fnc_handleMissionEnd.sqf`. Then in that thread, we spawn another thread on line 29. Then we spawn a thread per player on the foreach loop. One thread too many imo.

It's supposed to be called, not spawned, forget to edit the usage.

And, yeah, that's the only real way to make this quick is to spawn threads because the callExtension command takes so long sometimes. This is what you asked me to do in Trello.

Reply:
Calling it is part of the problem as it ties it into the main thread. It absolutely needs to be spawned, so with that in mind, we can get rid of that first spawn. Then we'll see if this causes any bugs Wednesday.